cdc/cdc.cc

```diff
@@ -534,15 +573,20 @@ future<std::vector<mutation>> append_log_mutations(
         service::storage_proxy::clock_type::time_point timeout,
         service::query_state& qs,
         std::vector<mutation> muts) {
-    return get_streams(ctx, s->ks_name(), s->cf_name(), timeout, qs).then(
-            [ctx, s = std::move(s), muts = std::move(muts)]
-            (transformer::streams_type streams) mutable {
-        transformer trans(ctx, std::move(s), streams);
-        muts.reserve(2 * muts.size());
-        for(int i = 0, size = muts.size(); i < size; ++i) {
-            muts.push_back(trans.transform(muts[i]));
-        }
-        return std::move(muts);
+    auto mp = ::make_lw_shared<std::vector<mutation>>(std::move(muts));
+
+    return get_streams(ctx, s->ks_name(), s->cf_name(), timeout, qs).then([ctx, s = std::move(s), mp, &qs](::shared_ptr<transformer::streams_type> streams) mutable {
+        mp->reserve(2 * mp->size());
+        auto trans = make_lw_shared<transformer>(ctx, s, std::move(streams));
+        auto i = mp->begin();
+        auto e = mp->end();
+        return parallel_for_each(i, e, [ctx, &qs, trans, mp](mutation& m) {
```

Comment:
Something to think about: Could we start here too many things parallel if "mp" has many mutations? Is this a likely occurance, or usually mp has only a small number of mutations?

Reply:
We probably could. And this is something that the future will tell. But not something to prematurely solve. 
Most of the time, mutations will be order of 1 in length.